File: app/src/main/java/com/autel/drone/demo/MissionFinishActionType.java
package com.autel.drone.demo;

/**
 * 任务完成动作，返航，悬停，未知
 */
public enum MissionFinishActionType {
    UNKNOWN(-1),
    /** 返航 */
    GO_HOME(0),
    /** 悬停 */
    HOVER(1),
    /** 降落 */
    LAND(2),
    /** 停在最一个航点位置 */
    LAST(3);

    private int value;

    MissionFinishActionType(int i) {
        this.value = i;
    }

    public int getValue() {
        return value;
    }

    public static MissionFinishActionType find(int value) {
        switch (value) {
            case 0:
                return GO_HOME;
            case 1:
                return HOVER;
            default:
                return UNKNOWN;
        }
    }
}

File: app/src/main/java/com/autel/mission/kml/custom/MissionFinishActionType.kt
package com.autel.mission.kml.custom

/**
 * @Author create by AUTEL
 * @Date 2022/09/21 11:56
 *
 * 任务完成后相机动作 Aircraft action after finish mission
 */
enum class MissionFinishActionType(val value: Int) {
    /** 返航 */
    GO_HOME(0),

    /** 悬停 */
    HOVER(1),

    /** 降落 */
    LAND(2),

    /** 停在最后一个航点位置 */
    LAST(3),

    /** 未知 */
    UNKNOWN(-1),
}